openoa/utils/downloader.py

```diff
@@ -330,6 +340,200 @@ def get_era5_monthly(
+
+
+    Returns:
+        df(:obj:`dataframe`): A dataframe containing time series of the requested reanalysis
+            variables:
+            1. u_ms: the U component of wind speed at a height of 100 m in m/s.
+            2. v_ms: the V component of wind speed at a height of 100 m in m/s.
+            3. temperature_K: air temperature at a height of 2 m in Kelvin.
+            4. surf_pres_Pa: surface pressure in Pascals.
+
+    Raises:
+        ValueError: If the start_date is greater than the end_date.
+    """
+
+    logger.info("Please note access to ERA5 data requires registration")
+    logger.info("Please see: https://cds.climate.copernicus.eu/api-how-to")
+
+    # set up cds-api client
+    try:
+        c = cdsapi.Client()
+    except SSLError:
+        print("Skipping certificate verification")
+        c = cdsapi.Client(verify=False)  # verification error for self-signed certificate
+
+    # create save_pathname if it does not exist
+    save_pathname = Path(save_pathname).resolve()
+    if not save_pathname.exists():
+        save_pathname.mkdir()
+
+    # get the current date
+    now = datetime.datetime.now()
+
+    # assign end_year to current year if not provided by the user
+    if end_date is None:
+        end_date = f"{now.year}-{now.month:02}-{now.day:02}"
+
+    # convert dates to datetime objects
+    start_date = datetime.datetime.strptime(start_date, "%Y-%m-%d")
+    end_date = datetime.datetime.strptime(end_date, "%Y-%m-%d")
+    end_date += datetime.timedelta(hours=23, minutes=59)  # include all times in last day
+
+    # check that the start and end dates are the right way around
+    if start_date > end_date:
+        logger.error("The start_date should be less than or equal to the end_date")
+        logger.error(f"start_date = {start_date.date()}, end_date = {end_date.date()}")
+        raise ValueError("The start_date should be less than or equal to the end_date")
+
+    # list all years that will be downloaded
+    years = list(range(start_date.year, end_date.year + 1))
+
+    # find the nearest coordinate grid point
+    node_spacing = 0.25
+    lat_nearest = node_spacing * np.round(lat / node_spacing)
+    lon_nearest = node_spacing * np.round(lon / node_spacing)
+
+    # Get data for single grid point
+    # See: https://cds.climate.copernicus.eu/datasets/reanalysis-era5-single-levels?tab=form
+    # for formulating other requests from cds
+    cds_dataset = "reanalysis-era5-single-levels"
+    cds_request = {
+        "product_type": "reanalysis",
+        "format": "netcdf",
+        "variable": [
+            "100m_u_component_of_wind",
+            "100m_v_component_of_wind",
+            "2m_temperature",
+            "surface_pressure",
+        ],
+        "year": None,
+        "month": None,
+        "day": [f"{i:02d}" for i in range(1, 32)],
+        "time": [f"{i:02d}:00" for i in range(24)],
+        "product_type": "reanalysis",
+        "area": [
+            lat_nearest + 0.01,
```

Comment:
What is the 0.1 buffer for? Unless something has changed, I believe a single point can be requested by repeating the latitude and longitude.

Reply:
You're right, the buffer doesn't seem to be needed.